core/data_processor.py

```
@@ -18,70 +37,150 @@ def __init__(self):
+                                    version = parts[1].strip()
+                                    # Use the canonical name from SUPPORTED_INSTRUMENTS
+                                    result['name'] = supported
+                                    result['version'] = version
+                                    result['detection_method'] = 'version_pattern'
+                                    logger.info(f"Detected instrument type (version pattern): {supported} v{version}")
+                                    break
+                
+                # After scanning all lines, if we found something, return it
+                if result['name'] != 'Unknown':
+                    self.instrument_info = result
+                    return result
+                
+                # Strategy 3: Parse from filename as fallback
+                filename = os.path.basename(file_path)
+                filename_upper = filename.upper()
+                
+                for prefix, instrument_name in FILENAME_PREFIX_MAP.items():
+                    if filename_upper.startswith(prefix.upper()):
+                        result['name'] = instrument_name
+                        result['detection_method'] = 'filename_pattern'
+                        logger.info(f"Detected instrument type (filename): {instrument_name}")
+                        self.instrument_info = result
+                        return result
                 
-                # If we get here, we successfully read the file but didn't find the instrument type
-                logger.info(f"Instrument type not found in first {max_lines} lines of file")
-                self.instrument_type = "Unknown"
-                return "Unknown"
+                logger.info(f"Instrument type not found in first {max_lines} lines or filename")
+                self.instrument_info = result
+                return result
                 
             except UnicodeDecodeError:
-                # Try next encoding
```

this isn't a bad comment, it adds clarification for why this error is ignored - is there a reason you cut it?